kong/db/schema/entities/upstreams.lua

```diff
@@ -212,6 +213,12 @@ local r =  {
 
     -- different headers
     { distinct = { "hash_on_header", "hash_fallback_header" }, },
+
+    -- if algorithm is "least" then hash_on must be "none"
```

Comment:
Why it must be none? Why cannot `least` just ignore it?

Reply:
I understand @Tieske's motivation for adding this constraint, but I think @bungle has a point there because there are already other fields that only make sense for certain configurations (e.g. `hash_on_cookie` only makes sense to be set if `hash_on == "cookie"`; we check that `hash_on_cookie` is required when hash mode is "cookie", but we don't check that it is _absent_ when hash mode is _not_ "cookie"). 

In the end, what @Tieske is really trying to represent with all those conditionals is a [tagged union](https://en.wikipedia.org/wiki/Tagged_union#1970s_&_1980s), but our schema library has not reached the state-of-the-art of the 1970s yet. ;)

Reply:
I don't mind leaving it out. Only from a usability prespective, when a user selects "least" and provides a "hash_on" value, he might expect a consistent-hashing approach to be used. An error would make it explicit that it isn't supported.

Reply:
but on the otherhand makes switching algorithms back and forth cumbersome (if using other than none).

Reply:
but on the otherhand makes switching algorithms back and forth cumbersome (if using other than none).